db/migrate/20170530073423_devise_create_users.rb

```diff
@@ -0,0 +1,46 @@
+class DeviseCreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+      t.string :username, null: false, default: ""
+      t.string :first_name, null: false, default: ""
+      t.string :last_name, null: false, default: ""
```

Comment:
Adding `default: ''` for `username`, `first_name` and `last_name` is dangerous. It means I can go to your user registration form, leave all the fields blank and submit, and a blank user will now be in the database. I recommend removing the `default: ''`